pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -62,8 +62,13 @@ func newNodeTree(nodes []*v1.Node) *NodeTree {
 	nt := &NodeTree{
 		tree: make(map[string]*nodeArray),
 	}
+	if len(nodes) == 0 {
+		return nt
+	}
```

Comment:
we don't need this check, the code below is generic enough to return an empty nt if nodes is empty.